### PR TITLE
Persian (Farsi) translation for Drupal Console book

### PR DESCRIPTION
**Pull request by novid**
Hello, this is the _complete_ **Persian (fa_IR)** translation of Drupal Console Documentation with the _exception_ of individual command translations which are better in original language.

### CODE REVIEW

**Comment by harold20** [comment]
Hello, @novid, thanks for the PR, but we manage the languages ​​in different repositories and this way we generate this documentation. If you want to help this is the repository https://github.com/hechoendrupal/drupal-console-fa

**Comment by novid** [comment]
HI @harold20, i would try to synchronize with the new fork and let you know in this issue when it's done.

**Comment by harold20** [comment]
Thanks @novid

**Comment by novid** [comment]
@harold20 I wanted to contribute [Drupal Console Documentation](https://hechoendrupal.gitbooks.io/drupal-console/content/en/index.html) not Drupal Console Commands which you mentioned.

Note that there is no help for contributors in [this project](https://github.com/hechoendrupal/drupal-console-book), why is that?

**Comment by enzolutions** [comment]
@novid let me rephrase what @harold20  tried to said.

1. The Drupal Console Book includes general instructions for Drupal Console, including command, usage, but command usage is in other section (explained in item # 2.
2. The commands translation part inside the Drupal Console Book is autogenerated using the language repository for each language, so you need to translate that part here.
3. To contribute with commands translation you need to do inside the specific language, long time ago I create a repo for Farsi translation, but the person interested at that moment didn't have enough time to do it. If you are interested, we can sync the latest version on https://github.com/hechoendrupal/drupal-console-fa, and you can continue from here.

About your second question, what kind of help are you looking for in this project?

Please, let me know if you understand me, if not let me know what part is not clear, we want to have our first RTL language supported.

If you are using the Drupal Slack, try to ping me inside the channel **#drupal-console**

**Comment by novid** [comment]
@enzolutions I just wanted to translate [Drupal Console Book](https://github.com/hechoendrupal/drupal-console-book) which is available in 12 different languages other than English. I **DO NOT** want to translate Drupal Console commands and their usage because the terminal emulators (bash, csh, zsh, ...) are designed specifically for LTR languages NOT RTL ones like _Arabic, Persian or Hebrew_.

So, [this repository](https://github.com/hechoendrupal/drupal-console-fa) only contains **yaml** for commands usage but not **Markdown** for documentation translation.

**Comment by novid** [comment]
@enzolutions I coordinated and translated [Drupal 8 User Guide](https://www.drupal.org/fa/docs/user_guide/fa/index.html) in just **3 month**! Persian (Farsi) language was the [first](https://twitter.com/eojthebrave/status/916410931873304576) one besides English that provides documentation for Drupal 8! But here, it is nearly **18 month** since the _initial and complete translation of Drupal Console Book in just 10 commits_ and we're arguing about how to do that.

If you still don't believe me, you can have a look at [my projects](http://novid.name/#projects). It's frustrating and makes me think i spent my free time on a wrong project ...

**Comment by enzolutions** [comment]
Hi @novid 

OK let do this.

1. We need to remove from the PR the files and folder fa-ir/commands/ because as I tell you in other GitHub issue is auto-generated and came from https://github.com/hechoendrupal/drupal-console-fa

After your remove it, I will merge and maybe you could contribute in the other repo https://github.com/hechoendrupal/drupal-console-fa, if you don't have time to do that is ok, just let me know and what I will do is merge and the auto-generator will overwrite with English translations. I just let you know the reason for the overwrite or delete, nothing related to the quality of your job.

2. I saw that you use fa-ir, but looking in the https://www.drupal.org/fa/docs/user_guide/fa/index.html the iso code used is **fa** and checking in wiki page https://en.wikipedia.org/wiki/Persian_language
the ISO 639-1 is **fa**. Again you are the one who knows your language I just need to know which one is better and to the changes in the repo https://github.com/hechoendrupal/drupal-console-fa.

Thank you so much for your contributions, and so sorry I make you feel that your contributions werent' appreciated.

**Comment by enzolutions** [comment]
@novid  thanks for delete the commands*

Could you define if we will use **fa** or **fa-ir**  which one is more natural for Persian speakers?

Could you also fix the conflict in LANGS.md  after defining which one is better?

Thanks again for all your effort.

**Comment by novid** [comment]
@enzolutions We can use **fa-ir** (Iran) instead of **fa** (General) because Farsi speakers in region (_Afghanistan & Tajikistan_) understand our language but have major differences in speaking & writing. If someday one of them wants to contribute in this project, they can use their language code respectively.

**Comment by enzolutions** [comment]
@novid 

Thanks for the clarification about **fa-ir**, we are ready to merge, right?

I will rename the repo https://github.com/hechoendrupal/drupal-console-fa and ask to @harold20  to sync with the latest version of English strings.

**Comment by novid** [comment]
@enzolutions Ok, that sounds great! Thank you for participating Eduardo

**Comment by enzolutions** [comment]
Thank you @novid  for your contribution

**Comment by enzolutions** [comment]
@novid At we are migrating our site and docs to https://www.gatsbyjs.org/, and at the same time, we will introduce support for Farsi Iranian (Persian)  you could see the preview https://drupal-console.web.app/docs/

I guess the orientation of the language is not correct, Do you are in the Drupal slack to talk about some tips to fix the website and other matters.

Screen Shot 2019-07-09 at 10 22 44 am
Screen Shot 2019-07-09 at 10 22 19 am


BTW we already promoted your work in twitter https://twitter.com/drupalconsole and in my facebook :P

**Comment by novid** [comment]
@enzolutions I'm not familiar with **GatsbyJS** or how it's presentation layer works, but if it's just s simple **CSS** file i can convert it to _RTL_.

**Comment by enzolutions** [comment]
@novid I think you need to explain to us what is wrong with current render ( if it is wrong), and tell us how is wrong, and how to fix, maybe with some documentation in English we are not familiar with RTL languages, from there we could work on it.

Thanks in advance for your help.

**Comment by enzolutions** [comment]
@novid  you could check a new update on https://drupal-console.web.app/docs/fa-ir/chains/what-is-a-chain-command

Please let know how looks for you.

Kind regards.

**Comment by novid** [comment]
@enzolutions That's pretty good except that the whole page layout should be RTL not just the text. For example, in the LTR version if the order is **sidebar (1/4)** then **content (3/4)** in RTL it should keep it the same as before, which means from _right to left of screen_ user would see sidebar first then content second.

In order to fully convert CSS styles you can use [this tool](https://rtl.daskhat.ir/).

**Comment by enzolutions** [comment]
@novid could you please check on https://drupal-console.web.app/docs/fa-ir/ to see if now looks better ? thanks in advance.

**Comment by novid** [comment]
@enzolutions It looks great! thank you :)
